### PR TITLE
add more integration test cases

### PR DESCRIPTION
**Pull request by carmark**

### CODE REVIEW

File: pkg/kubectl/cmd/run.go [hunk 1]
```diff
@@ -129,6 +130,15 @@ func Run(f *cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *cob
 	if err != nil {
 		return err
 	}
+	client, err := f.ClientConfig()
+	if err != nil {
+		return err
+	}
+	if strings.Index(client.Host, "http://") == 0 {
```
Comment (feiskyer) [inline pkg/kubectl/cmd/run.go]: Could this process move to api instead?

File: pkg/kubectl/cmd/delete.go [hunk 2]
```diff
@@ -110,6 +111,15 @@ func RunDelete(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 	if err != nil {
 		return err
 	}
+	client, err := f.ClientConfig()
```
Comment (feiskyer) [inline pkg/kubectl/cmd/delete.go]: Could this process move to api instead?

File: pkg/kubectl/cmd/autoscale.go [hunk 3]
```diff
@@ -73,6 +73,10 @@ func RunAutoscale(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []
 	if err != nil {
 		return err
 	}
+	tenant, enforceTenant, err := f.DefaultTenant()
```
Comment (feiskyer) [inline pkg/kubectl/cmd/autoscale.go:76]: Could this process move to api instead?

File: hack/test-keystone.sh [hunk 4]
```diff
@@ -313,31 +314,350 @@ runTests() {
...
   kube_flags=${kube_user_flags[@]}
   kube::test::get_object_assert "namespaces" "{{range.items}}{{$id_field}}:{{end}}" ''
 
   # Create 'test' tenant
-  # kubectl create -f docs/user-guide/te.yaml "${kube_admin_flags[@]}"
+  kubectl create -f docs/user-guide/te.yaml "${kube_admin_flags[@]}"
   # Create 'test' namespace for 'test' tenant
-  # kubectl create -f docs/user-guide/ns.yaml "${kube_user_flags[@]}"
-  # kube::test::get_object_assert "namespaces" "{{range.items}}{{$id_field}}:{{end}}" 'test:'
+  kubectl create -f docs/user-guide/ns.yaml "${kube_user_flags[@]}"
```
Comment (feiskyer) [inline hack/test-keystone.sh:340]: Those tests are same with test-cmd.sh. Here should create namespace in test tenant and ensure new resources created properly.

File: hack/test-keystone.sh [hunk 5]
```diff
@@ -313,31 +314,350 @@ runTests() {
...
   ###########################
-
   kube::log::status "Testing kubectl(${version}:pods)"
 
   kube_flags=${kube_admin_flags[@]}
   ### Create POD valid-pod from JSON
   # Pre-condition: no POD is running
   kube::test::get_object_assert "pods" "{{range.items}}{{$id_field}}:{{end}}" ''
+  # Create 'test' namespace for 'test' tenant
+  kubectl create -f docs/user-guide/ns.yaml "${kube_user_flags[@]}"
```
Comment (feiskyer) [inline hack/test-keystone.sh:360]: Here is the same as above

File: hack/test-keystone.sh (hunk 4 above)
Comment (carmark) [inline hack/test-keystone.sh:340]: Do you think which tenant is the created namespace belonged to?  I think it is a namespace of `test` tenant.  I don't think those tests are same as the test-cmd.sh, they have different ways to access the resource.

Comment (feiskyer) [inline hack/test-keystone.sh:340]: 1. There should be tests to assert `tenant` of created resources
2. `kube::test::get....` should also query resources in dedicated tenant

Comment (carmark) [inline hack/test-keystone.sh:340]: Line 340~343

```
+  kubectl create -f docs/user-guide/ns.yaml "${kube_user_flags[@]}"
+  kube::test::get_object_assert "namespaces" "{{range.items}}{{$id_field}}:{{end}}" 'test:'
+  kube_flags=${kube_admin_flags[@]}
+  kube::test::get_object_assert "namespaces" "{{range.items}}{{$id_field}}:{{end}}" 'default:test:'
```

Comment (feiskyer) [inline hack/test-keystone.sh:340]: ok

**Pull request merged**